internal/zinc-compile-core/src/main/scala/sbt/internal/inc/ReporterManager.scala

```diff
@@ -0,0 +1,102 @@
+package sbt.internal.inc
+
+import java.io.{ OutputStreamWriter, PrintStream, PrintWriter }
+import java.nio.charset.StandardCharsets
+import java.util.logging.Level
+
+import sbt.internal.util.{ ConsoleAppender, MainAppender, ManagedLogger }
+import sbt.util.LogExchange
+import sbt.util.{ Level => SbtLevel }
+import xsbti.{ Position, Reporter, ReporterConfig }
+
+import scala.util.matching.Regex
+
+object ReporterManager {
+  import java.util.concurrent.atomic.AtomicInteger
+  private val idGenerator: AtomicInteger = new AtomicInteger
+  private val DefaultName = "zinc-out"
+  private def generateZincReporterId(name: String): String =
+    s"$name-${idGenerator.incrementAndGet}"
+
+  /**
+   * Returns the id that is used to know what's the minimum level from which the
+   * reporter should log messages from the compilers.
+   *
+   * It uses `java.util.logging.Level` because `sbt.util.Level` is a Scala enumeration
+   * and there are lots of deficiencies in its design (as well as in some bugs in
+   * scalac) that prevent its leak in the public Java API, even though it's supposed
+   * to be public. For instance, I have tried to refer to it in the following ways:
+   *
+   * - `sbt.util.Level.Info` -> Scalac passes, Javac fails.
+   * - `sbt.util.Level$.Info` -> Scalac error: type mismatch.
+   * - `sbt.util.Level$.MODULE$.Info` -> Scalac fails with stable identifier required.
+   *
+   * Using `CompileOrder.Mixed` does not solve the issue. Therefore, we fallback
+   * on `java.util.logging.Level` and we transform to `sbt.util.Level` at runtime. This
+   * provides more type safety than asking users to provide the id of `sbt.util.Level`.
+   *
+   * TODO(someone): Don't define `sbt.util.Level` as a Scala enumeration.
+   */
+  private def fromJavaLogLevel(level: Level): SbtLevel.Value = {
+    level match {
+      case Level.INFO    => SbtLevel.Info
+      case Level.WARNING => SbtLevel.Warn
+      case Level.SEVERE  => SbtLevel.Error
+      case Level.OFF     => sys.error("Level.OFF is not supported. Change the logging level.")
+      case _             => SbtLevel.Debug
+    }
+  }
```

Comment:
I think it's too big of a change, doesn't give us much... historically we've always used sbt's level logging and I think it's good to honor that tradition. This is a workaround for a technical Java-Scala compatibility issue.